File: hw/paillacho/U3/HW20Composite/Composite/src/main/java/ec/edu/espe/composite/model/Clerk.java
package ec.edu.espe.composite.model;

/**
 *
 * @author Carlos Paillacho
 */
public class Clerk extends Employee {

    /**
     * Default constructor sets the title.
     */
    public Clerk() {
        title = "Clerk";
    }

    /**
     * Constructs a Clerk with a name.
     *
     * @param aName the clerk's name
     */
    public Clerk(String aName) {
        this();
        name = aName;
    }

    @Override
    public void stateName() {
        super.stateName();
    }
}

File: hw/bonilla/U3/HW20Composite/Compositejs/model/Clerk.js
const Employee = require('./Employee');

class Clerk extends Employee {
  constructor(aName) {
    super();
    this.title = "Clerk";
    if (aName) this.name = aName;
  }

  stateName() {
    super.stateName();
  }
}

module.exports = Clerk;
